src/main/java/org/xbill/DNS/Header.java

```diff
@@ -31,15 +31,14 @@
    * @param id The message id
    */
   public Header(int id) {
-    this();
-    setID(id);
+    counts = new int[4];
+    flags = 0;
+    this.id = id;
```

Comment:
This now allows out-of-range IDs. Since calling setID from a constructor isn't really a good idea (virtual member call), please either copy the range check or extract it into a private method.

Reply:
sure thing!